### PR TITLE
018rc2

### PR DESCRIPTION
**Pull request by amueller**

### CODE REVIEW

**Comment by amueller** [comment]
@jakirkham after circleci ran through, shouldn't this how up here? https://anaconda.org/conda-forge/repo?label=rc

**Comment by amueller** [comment]
> No BINSTAR_TOKEN present, so no upload is taking place. The distribution just built is not already available on conda-forge's rc channel.

Is that because It's a PR? I'll merge and we find out I guess ;)

**Comment by amueller** [comment]
worked! https://anaconda.org/conda-forge/scikit-learn/labels

**Comment by jakirkham** [comment]
🎉 Glad it worked.

Yes deployments don't happen without adding commits to the feedstock directly. Merge or otherwise.

**Comment by amueller** [comment]
Thank you so much for your help and patience @jakirkham :)

It would be great if there was another workaround for the long build than disabling the tests ;)

File: recipe/meta.yaml [hunk 1]
```diff
@@ -44,7 +44,7 @@ test:
     - sklearn
 
   commands:
-    - nosetests sklearn --exe
+    - echo "this is the best test ever"
```
Comment (jakirkham) [inline recipe/meta.yaml]: 😆

**Comment by jakirkham** [comment]
Of course. Glad to help.

Hope this experience with conda-forge was positive. Certainly we have some more kinks to work out.

Yeah we had mentioned dropping NumPy 1.10 generally. Though I don't think we ever resolved that. Admittedly that isn't a great fix as we will inevitably need to support 2 NumPy versions again once 1.12 comes out just to handle packages caught in the transition.

Looks like it was just CircleCI. It doesn't really have support for a proper build matrix like Travis CI hence why we do all the builds in serial. They do have something called parallelism, which can be driven with a matrix. However, it has some drawbacks too mainly in terms of the complexity to set it up and some potential UX issues. That being said that is the best long term option. See issue ( https://github.com/conda-forge/conda-smithy/issues/203 ) for more details.

**Comment by jakirkham** [comment]
Sorry meant to add more to that comment.

The more practical short term (and potentially long term) solution is to split out the 2 NumPy build matrices into 2 branches. This adds some maintenance overhead. However, that can be seen as a positive as it encourages the eventual drop of old NumPy versions. We had to do this for SciPy not so long ago for the same reasons. Please see PR ( https://github.com/conda-forge/scipy-feedstock/pull/11 ) for an example. I think it would be very practical and effective to do the same thing here.

**Comment by jakirkham** [comment]
Broke this out as a new comment as it is a bit different.

Eventually we do want our own build infrastructure as CIs have all sorts of annoying limitations. This time limit is merely one example. There are just a lot of moving parts in solving that problem. Though we have heard rumors that scikit-learn gets some build time at Rackspace. Is that true? Do you or does anyone on the project know about this and/or have any advice for us?

**Comment by ogrisel** [comment]
>  Though we have heard rumors that scikit-learn gets some build time at Rackspace. Is that true? Do you or does anyone on the project know about this and/or have any advice for us?

We have a rackspace cloud account to support the development of scikit-learn and related open source projects. It's the equivalent of $2000 per month but it wont be enough to run CI for all the projects on conda-forge.

At the moment we only use it to publish binary wheels for the scipy stack (as a staging step for CI before publishing to PyPI) and the bandwith costs are already in the order of $200 per month IIRC.

**Pull request merged**